ui/app/src/state/reducers/preview.ts

```diff
@@ -238,6 +241,23 @@ const fetchGuestModelsCompleteHandler = (state, { type, payload }) => {
 };
 
 const reducer = createReducer<GlobalState['preview']>(initialState, {
+  [initPreviewConfig.type]: (state, { payload }) => {
+    const configDOM = fromString(payload.configXml);
+    const previewConfigEl = configDOM.querySelector('[id="craftercms.components.Preview"]');
+    const initialEditModeOn = previewConfigEl?.getAttribute('initialEditModeOn');
+    const initialHighlightMode = previewConfigEl?.getAttribute('initialHighlightMode');
+    const xbDetectionTimeoutMs = parseInt(previewConfigEl?.getAttribute('xbDetectionTimeoutMs'));
+
+    state.editMode = payload.storedEditMode ?? (initialEditModeOn ? initialEditModeOn === 'true' : state.editMode);
+    state.highlightMode =
+      payload.storedHighlightMode ??
+      (['all', 'move'].includes(initialHighlightMode) ? initialHighlightMode : state.highlightMode);
+    state.editModePadding = payload.storedPaddingMode ?? state.editModePadding;
+    state.xbDetectionTimeoutMs =
+      Boolean(xbDetectionTimeoutMs) && xbDetectionTimeoutMs > 0
```

Comment:
`0` should be a valid value which means to not execute the check.
```suggestion
      nnou(xbDetectionTimeoutMs)
```